spec/ref_arch_setup/download_helper_spec.rb

```diff
@@ -0,0 +1,933 @@
+        expect(subject).not_to receive(:puts).with(result)
+
+        expect(test_versions_result).to receive(:text).and_return("")
+        expect { subject.ensure_valid_prod_version(version) }.to raise_error(RuntimeError)
+      end
+
+      it "raises the expected error" do
+        error = "Specified version not found: #{version}"
+        allow(subject).to receive(:puts)
+        expect(subject).to receive(:parse_prod_versions_url).and_return(test_versions_result)
+        expect(test_versions_result).to receive(:text).and_return("")
+        expect { subject.ensure_valid_prod_version(version) }.to raise_error(RuntimeError, error)
+      end
+    end
+  end
+
+  describe "#ensure_supported_prod_version" do
+    context "when the version is supported" do
+      version = "2018.1.4"
+      message = "Specified version #{version} is supported by RAS"
+
+      it "outputs a confirmation" do
+        expect(subject).to receive(:puts).with(message)
+        subject.ensure_supported_prod_version(version)
+      end
+
+      it "returns true" do
+        allow(subject).to receive(:puts)
+        expect(subject.ensure_supported_prod_version(version)).to eq(true)
+      end
+    end
+
+    context "when the version is not supported" do
+      version = "2017.1.4"
+      message = "The minimum supported version is #{subject::MIN_PROD_VERSION}"
+      error = "Specified version #{version} is not supported by RAS"
+
+      it "outputs an explanation (and raises an error)" do
+        expect(subject).to receive(:puts).with(message)
+        expect { subject.ensure_supported_prod_version(version) }
+          .to raise_error(RuntimeError)
+      end
+
+      it "raises the expected error" do
+        allow(subject).to receive(:puts)
+        expect { subject.ensure_supported_prod_version(version) }
+          .to raise_error(RuntimeError, error)
+      end
+    end
+  end
+
+  describe "#fetch_prod_versions" do
+    context "when called" do
+      it "returns the default result from parse_prod_versions_url" do
+        expect(subject).to receive(:parse_prod_versions_url).and_return(test_versions_result)
+        expect(subject.fetch_prod_versions).to eq(test_versions_result)
+      end
+    end
+  end
+
+  describe "#parse_prod_versions_url" do
+    let(:test_net_http) { Class.new }
+    let(:test_uri) { Class.new }
+    let(:test_oga) { Class.new }
+
+    let(:test_http_response) { Class.new }
+    let(:test_body) { Class.new }
+    let(:test_document) { Class.new }
+    let(:test_result) { Class.new }
+
+    message = "Checking Puppet Enterprise Version History: #{TEST_PE_VERSIONS_URL}"
+
+    before do
+      stub_const("Net::HTTP", test_net_http)
+      stub_const("URI", test_uri)
+      stub_const("Oga", test_oga)
+    end
+
+    context "when the response is valid" do
+      context "when an xpath is specified" do
+        it "uses the specified xpath and returns the result" do
+          xpath = "//table/tbody/tr[1]/td[1]"
```

Comment:
If I read this right.  The passed in xpath and the default xpath are the same in these tests.  So the test doesn't ensure it used the passed in one, as it could have used the default.

Reply:
The default is "//table/tbody/tr/td[1]" (all rows); this test uses "//table/tbody/tr[1]/td[1]" (first row).

Reply:
I looked like 3 times... didn't see the difference... lol  Now I do. :)